packages/rpc-core/src/rpc-methods/__typetests__/get-block-type-test.ts

```diff
@@ -0,0 +1,1256 @@
+import { Base58EncodedAddress } from '@solana/addresses';
+import { Blockhash, TransactionVersion } from '@solana/transactions';
+import assert from 'assert';
+
+import { TransactionError } from '../../transaction-error';
+import {
+    Base58EncodedBytes,
+    Base58EncodedDataResponse,
+    Base64EncodedDataResponse,
+    LamportsUnsafeBeyond2Pow53Minus1,
+    Reward,
+    TokenBalance,
+    TransactionStatus,
+    U64UnsafeBeyond2Pow53Minus1,
+} from '../common';
+import { mockTypeTestResponse, mockTypeTestRpc } from './common';
+
+const rpc = mockTypeTestRpc();
+
+function assertBase(
+    response: {
+        blockhash: string;
+        blockHeight: bigint;
+        blockTime: number;
+        parentSlot: bigint;
+        previousBlockhash: string;
+    } & { [key: string]: unknown }
+) {
+    response.blockhash satisfies string;
+    response.blockHeight satisfies bigint;
+    response.blockTime satisfies number;
+    response.parentSlot satisfies bigint;
+    response.previousBlockhash satisfies string;
+}
+
+// First overload
+// Rewards set to `false`
+{
+    const response = mockTypeTestResponse(
+        rpc.getBlock(0n, {
+            // No extra configs
+            rewards: false,
+            transactionDetails: 'none',
+        })
+    );
+    assert(response);
+    assertBase(response);
+    // @ts-expect-error Should be undefined
+    response.transactions satisfies Array<unknown>;
```

Comment:
I wish there was a better way to assert ‘is not a property of X.’ I looked for inspiration [here](https://github.com/microsoft/TypeScript-DOM-lib-generator/tree/main/unittests/files) but didn't find any.

How about this? https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAQwM6oKYCcoDk5QCCAClnAA7ZQCeAPACqIYAeUGYAJqonAEYBWGaABpE9UhVzIAthiat2XRKihYYYAOYA+ABQB9AFyIA3gG0J5ROsQBrDNTjAxAXSMX5bTt3FlyU2YgA-IhgGABu2IhG9Oa+zgC+onrkvv4Y0RZpAJRGYXAwHCbxAFDFEAgqKIgAvCYAFhgANo1wRgDkUA1YGG0lAPR9iAACUKgAtCyU0BNYZFjFaJg4+EQWVNQ6yKJtDc1wbVkA3AvoVCskvuub2xpwcBy81D1HQA

Reply:
Sure that works!

Reply:
Weird, that works in playground, but not in my local setup.

How about this?

```typescript
response satisfies Omit<unknown, 'transactions' | 'rewards'>;
```

Reply:
Nope, nvm doesn't work

Reply:
Oh wait, I had yours backwards. It works